tests/integration/userCompanies.test.js

```diff
@@ -5,13 +5,149 @@ const { getTokenByCredentials, getToken } = require('./helpers/authentication');
+  beforeEach(populateDB);
+
+  describe('COACH', () => {
+    beforeEach(async () => {
+      authToken = await getToken('coach');
+    });
+
+    it('should create user company', async () => {
+      const payload = { user: usersSeedList[10]._id, company: authCompany._id };
+
+      const res = await app.inject({
+        method: 'POST',
+        url: '/usercompanies',
+        payload,
+        headers: { Cookie: `alenvi_token=${authToken}` },
+      });
+
+      expect(res.statusCode).toBe(200);
+    });
+
+    it('should return 403 if user is does not exists', async () => {
+      const payload = { user: new ObjectId(), company: authCompany._id };
+
+      const res = await app.inject({
+        method: 'POST',
+        url: '/usercompanies',
+        payload,
+        headers: { Cookie: `alenvi_token=${authToken}` },
+      });
+
+      expect(res.statusCode).toBe(403);
+    });
+
+    it('should return 404 if company is not client user company', async () => {
+      const payload = { user: usersSeedList[10]._id, company: otherCompany._id };
+
+      const res = await app.inject({
+        method: 'POST',
+        url: '/usercompanies',
+        payload,
+        headers: { Cookie: `alenvi_token=${authToken}` },
+      });
+
+      expect(res.statusCode).toBe(404);
+    });
+
+    it('should return 409 if user already have a company', async () => {
+      const payload = { user: usersSeedList[0]._id, company: authCompany._id };
+
+      const res = await app.inject({
+        method: 'POST',
+        url: '/usercompanies',
+        payload,
+        headers: { Cookie: `alenvi_token=${authToken}` },
+      });
+
+      expect(res.statusCode).toBe(409);
+      expect(res.result.message).toBe('Ce compte est déjà rattaché à une structure.');
+    });
+
+    it('should return good message if try to link user with 1 inactive company with wrong start date', async () => {
+      const payload = { user: usersSeedList[7]._id, company: authCompany._id, startDate: '2021-10-19T23:00:00.000Z' };
+
+      const res = await app.inject({
+        method: 'POST',
+        url: '/usercompanies',
+        payload,
+        headers: { Cookie: `alenvi_token=${authToken}` },
+      });
+
+      expect(res.statusCode).toBe(409);
+      expect(res.result.message).toBe('Ce compte est déjà rattaché à une structure jusqu\'au 20/11/2021.');
+    });
+  });
+
+  describe('TRAINING_ORGANISATION_MANAGER', () => {
+    beforeEach(async () => {
+      authToken = await getToken('training_organisation_manager');
+    });
+
+    it('should create user company', async () => {
+      const payload = { user: usersSeedList[10]._id, company: authCompany._id };
+
+      const res = await app.inject({
+        method: 'POST',
+        url: '/usercompanies',
+        payload,
+        headers: { Cookie: `alenvi_token=${authToken}` },
+      });
+
+      expect(res.statusCode).toBe(200);
+    });
+
+    it('should return 403 if company is does not exists', async () => {
```

Comment:
idem + ce test peut être dans coach, d'ailleurs j'aurais plus mis ROF comme cas par défaut et juste les cas spécifiques à coach dans coach

Reply:
je crois que tu n'as pas pris la 2e partie du retour

Reply:
il me semble que si :
- dans 'COACH', je n'ai plus que 2 tests "should return 404 if company is not client user company" et "should create usercompany"
- dans ROF, j'ai tous les autres
ou alors je n'ai pas compris ton retour ^^